core/src/main/scala/org/apache/spark/util/collection/WritablePartitionedPairCollection.scala

```diff
@@ -74,6 +74,31 @@ private[spark] object WritablePartitionedPairCollection {
     }
   }
 
+  /* Takes an optional parameter (keyComparator), use if provided
+   * and returns a comparator for the partitions
+   */
+  def getComparator[K](keyComparator: Option[Comparator[K]]) : Comparator[(Int, K)] = {
+    val comparator : Comparator[(Int, K)] =
+      if (keyComparator.isEmpty) {
```

Comment:
`isDefined` is probably a tiny bit more conventional (and then flip the logic here of course)